### PR TITLE
Explain Glide in developer documentation

### PR DESCRIPTION
**Pull request by kevgo**
Our developer documentation should explain how to use Glide properly, since that is not immediately obvious, and it's part of our tool chain.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -46,6 +48,15 @@ This guide will help you get started and outline some things you should know whe
 * See [here](./documentation/development/testing.md) for how to run the tests
 
 
+## Developing
+
+* update depependencies: `glide up`
+* add a new Go library:
+  * update dependencies in a separate PR
+  * `glide get <package name>`
+  * add the source code of the added library, which is located in [vendor](vendor), to Git
```
Comment (charlierudolph) [inline CONTRIBUTING.md]: Do you have to do anything special here? Or is this just letting the developer know that we check in the dependencies?

Comment (kevgo) [inline CONTRIBUTING.md]: Yes, tell them that they should check in the source code of the dependency. Some Go projects don't do that, but given GT's long-term aim, I think we should, and developers should know that.

Comment (charlierudolph) [inline CONTRIBUTING.md]: Thats not clear to me. Thoughts on mentioning that dependencies are checked into source control in the vendor folder as the first bullet above update dependencies?

Then this bullet would simple be commit the updated glide.yaml / glide.lock / vendor folder.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -46,6 +48,15 @@ This guide will help you get started and outline some things you should know whe
 * See [here](./documentation/development/testing.md) for how to run the tests
 
 
+## Developing
+
+* update depependencies: `glide up`
```
Comment (charlierudolph) [inline CONTRIBUTING.md]: I believe `dependencies` is misspelled

Comment (kevgo) [inline CONTRIBUTING.md]: correct! fixed

File: CONTRIBUTING.md (hunk 1 above)
Comment (kevgo) [inline CONTRIBUTING.md]: Fixed this. What do you think?

**Comment by kevgo** [comment]
All comments addressed

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -46,6 +48,18 @@ This guide will help you get started and outline some things you should know whe
...
+## Developing
+
+* all dependencies are located in the [vendor](vendor) folder,
+  which is checked into Git
+* update dependencies: `glide up`
+* adding a new Go library:
+  * update dependencies in a separate PR
+  * `glide get <package name>`
+  * don't forget to add the updated glide files and vendor folder
+    in the pull request for the feature that requires the new library
```
Comment (charlierudolph) [inline CONTRIBUTING.md]: I doubt its really a matter of remembering. Aren't we are trying to let a user know that updating a dependency will result in changes to the vendor folder? Maybes something like
```
* your PR should contain updates to the glide files and vendor folder
``

**Pull request merged**